cob_scan_unifier/src/scan_unifier_node.cpp

```diff
@@ -123,150 +166,99 @@ void scan_unifier_node::getParams()
     config_.number_input_scans = 0;
     ROS_ERROR("No parameter input_scans on parameter server!! Scan unifier can not subscribe to any scan topic!");
   }
-}
 
-/**
- * @function getLoopRate
- * @brief getter function for the loop rate
- *
- * input: -
- * output:
- * @return the loop rate
- */
-double scan_unifier_node::getLoopRate()
-{
-  return config_.loop_rate;
+  if(!pnh_.hasParam("frame"))
+  {
+    ROS_WARN("No parameter frame on parameter server. Using default value [base_link].");
+    frame_ = "base_link";
+  }
+  else
+  {
+    pnh_.getParam("frame", frame_);
 }
-
-/**
- * @function set_new_msg_received
- * @brief setter function for new_msg_received variable
- *
- * input:
- * @param message received information
- * @param scan id to find right scan in struct
- * output: -
- */
-void scan_unifier_node::set_new_msg_received(bool received, int scan_id)
-{
-  pthread_mutex_lock(&m_mutex);
-  vec_laser_struct_.at(scan_id).new_msg_received = received;
-  pthread_mutex_unlock(&m_mutex);
 }
 
-/**
- * @function get_new_msg_received
- * @brief getter function for new_msg_received variable for checking wether a new msg has been received, triggering publishers accordingly
- *
- * input:
- * @param scan id to find right scan in struct
- * output: -
- * @return the new_msg_received variable
- */
-bool scan_unifier_node::get_new_msg_received(int scan_id)
-{
-  pthread_mutex_lock(&m_mutex);
-  bool ret_val = vec_laser_struct_.at(scan_id).new_msg_received;
-  pthread_mutex_unlock(&m_mutex);
-  return ret_val;
-}
 
-/**
- * @function initLaserScanStructs
- * @brief initialize a vector of laser scan structs (member variable vec_laser_struct_) with a given number
- * (from parameter server, stored in config_ struct)
- *
- * input: -
- * output: -
- */
-void scan_unifier_node::initLaserScanStructs()
+void ScanUnifierNode::messageFilterCallback(const sensor_msgs::LaserScan::ConstPtr& first_scanner, const sensor_msgs::LaserScan::ConstPtr& second_scanner)
```

Comment:
I'd prefer `scan1` insead of `first_scanner`, `scan2` instead of `second_scanner`....